search/indexing_api.py

```diff
@@ -122,3 +143,98 @@ def increment_document_integer_field(doc_id, field_name, incr_amount):
         },
         update_key='script'
     )
+
+
+def index_post_with_comments(post_id):
+    """
+    Index a post and its comments
+
+    Args:
+        post_id (str): The post string
+    """
+    from channels.api import Api
+
+    conn = get_conn()
+    api_username = settings.INDEXING_API_USERNAME
+    client = Api(User.objects.get(username=api_username))
+    post = client.get_post(post_id)
+    comments = post.comments
+    # Make sure all morecomments are replaced before serializing
+    comments.replace_more(limit=None)
```

Comment:
I'm going to leave this since I'm not sure how to do that cleanly, if you have a suggestion let me know

Reply:
I was thinking something like: 

```python
def get_all_comments(self, post_id):
  post = self.get_post(post_id)
  comments = post.comments
  comments.replace_more(limit=None)
  return comments
```

Don't know if that does what we would need it to

Reply:
That function would need to return the post so that we could serialize it later in the function. I'm not sure how intuitive it would be to return a post from that function when `list_comments` returns a `CommentTree`